### PR TITLE
DOCFIX: Metrics

### PR DESCRIPTION
**Pull request by mr-ubik**
DOCFIX:
- All properties docstring now always say Retrieve and not Return
- Rewrote the docstring for the Classes in `ashpy.contexts`
- Add proper documentation and types to the `ashpy.metrics`
- Add intersphinx mapping for `numpy` docs

MISC:
- Add mypy.ini
- Renamed types.py to ashtypes
- Add `.coveragerc` to ignore `if TYPE_CHECKING:` lines

### CODE REVIEW

File: ashpy/metrics/metric.py [hunk 1]
```diff
@@ -14,53 +14,72 @@
...
+if TYPE_CHECKING:
+    from ashpy.contexts import BaseContext
 
 
 class Metric(ABC):
-    """Metric is the abstract class that every ash metric must implement."""
+    """
+    Metric is the abstract class that every ash Metric must implement.
+
+    TODO: (ASK FEDE) AshPy metrics act as wrapper for Keras ones.
```
Comment (EmanueleGhelfi) [inline ashpy/metrics/metric.py]: Have we solved this?

File: ashpy/metrics/gan.py [hunk 2]
```diff
@@ -13,43 +13,76 @@
...
         super().__init__(
             name="d_loss",
             metric=tf.metrics.Mean(name="d_loss", dtype=tf.float32),
             model_selection_operator=model_selection_operator,
             logdir=logdir,
         )
 
-    def result(self):
-        return self._metric.result().numpy()
+    # def result(self) -> np.ndarray:
```
Comment (EmanueleGhelfi) [inline ashpy/metrics/gan.py]: Do we need to keep this comments?

Comment (mr-ubik) [reply ashpy/metrics/gan.py]: We can delete these sections, they should be handled by the parent class.

File: ashpy/metrics/metric.py (hunk 1 above)
Comment (mr-ubik) [reply ashpy/metrics/metric.py]: Oops. My bad. This was resolved.

**Comment by EmanueleGhelfi** [comment]
I think the codecov service fails because of the TYPE_CHECKING part of the code. Maybe we can integrate this plugin in our pipeline: https://github.com/dbader/pytest-mypy

File: ashpy/metrics/gan.py (hunk 2 above)
Comment (mr-ubik) [reply ashpy/metrics/gan.py]: Solved by ccf5ed6

File: ashpy/metrics/metric.py (hunk 1 above)
Comment (mr-ubik) [reply ashpy/metrics/metric.py]: Solved by ccf5ed6

**Comment by EmanueleGhelfi** [comment]
@mr-ubik I've manually ignored the issue on codefactor related to the FIXME comment.

**Pull request merged**